src/core/src/main/java/org/apache/jmeter/util/keystore/JmeterKeyStore.java

```diff
@@ -303,36 +323,60 @@ private boolean isIndexInConfiguredRange(int index) {
      * @throws IllegalArgumentException if {@link #clientCertAliasVarName}
      *                                  is not empty and no key for this alias could be found
      */
-    public String getAlias() {
+    public String getAlias(String [] keyTypes) {
```

Comment:
I would suggest returning Optional<String> since we're touching this class

Reply:
this class should be removed though. I don't think it adds value :-/

Reply:
As @veselov  explained, it implements the ability to test client certificate authentication.
This feature is useful and needed, so maybe it requires refactoring but just removing it is not acceptable.